### PR TITLE
IBGU: Add FinalizeUpgrade, FinalizeRollback, and AbortOnFailure actions 

### PR DESCRIPTION
**Pull request by sudomakeinstall2**
- Introduced explicit actions for finalizeUpgrade, finalizeRollback, and
  AbortOnFailure. The selector is automatically added:
  - For FinalizeUpgrade, selects clusters that have completed the
    upgrade.
  - For FinalizeRollback, selects clusters that have completed the
    rollback.
  - For AbortOnFailure, selects clusters that have either failed
    prep or upgrade.

- Updated cluster label format from
  `lcm.openshift.io/ibgu-<ibguName>-<action>: completed/failed` to
`lcm.openshift.io/ibgu-<action>-completed/failed: ""`. With each cluster
uniquely associated with a single IBGU, specifying the IBGU name in the
label is now redundant.

### CODE REVIEW

File: controllers/imagebasedgroupupgrade.go [hunk 1]
```diff
@@ -120,15 +120,15 @@ func (r *IBGUReconciler) ensureClusterLabels(ctx context.Context, ibgu *ibguv1al
...
+			labelsToAdd[fmt.Sprintf("lcm.openshift.io/ibu-%s-failed", strings.ToLower(action.Action))] = ""
 		}
 		removeLabels := false
 		for _, action := range clusterState.CompletedActions {
 			if action.Action == ibguv1alpha1.Finalize || action.Action == ibguv1alpha1.Abort {
 				removeLabels = true
 				break
 			}
-			labelsToAdd[fmt.Sprintf("lcm.openshift.io/ibgu-%s-%s", ibgu.GetName(), action.Action)] = "completed"
+			labelsToAdd[fmt.Sprintf("lcm.openshift.io/ibu-%s-completed", strings.ToLower(action.Action))] = ""
```
[the remark this answers is not in the document]
Comment (sudomakeinstall2) [reply controllers/imagebasedgroupupgrade.go:131]: the Linter will complain if you repeat it more than two times.

**Review by jc-rh — APPROVED** [review]

**Pull request merged**